weaviate/users/sync.pyi

```diff
@@ -1,37 +1,41 @@
-from typing import Dict, List, Literal, Union, overload
+from typing import Any, Dict, Generic, List, Optional, Union, cast
+from httpx import Response
+from weaviate.connect import executor
+from weaviate.connect.v4 import _ExpectedStatusCodes, ConnectionType
+from weaviate.rbac.models import Role, RoleBase, WeaviateDBUserRoleNames, UserTypes
+from weaviate.users.users import USER_TYPE_DB, USER_TYPE_OIDC, USER_TYPE, UserDB, OwnUser
+from weaviate.util import _decode_json_response_dict
 from weaviate.connect.v4 import ConnectionSync
-from weaviate.users.executor import _DeprecatedExecutor, _DBExecutor, _OIDCExecutor
-from weaviate.users.users import UserDB, OwnUser
+from .executor import _BaseExecutor, _UsersExecutor, _UsersOIDCExecutor, _UsersDBExecutor
 
-from weaviate.rbac.models import Role, RoleBase
-from typing_extensions import deprecated
+class _Base(_BaseExecutor[ConnectionSync]):
+    def _get_roles_of_user(
+        self, user_id: str, user_type: USER_TYPE, include_permissions: bool
+    ) -> Union[Dict[str, Role], Dict[str, RoleBase]]: ...
+    def _get_roles_of_user_deprecated(
+        self, user_id: str
+    ) -> Union[Dict[str, Role], Dict[str, RoleBase]]: ...
+    def _assign_roles_to_user(
+        self, roles: List[str], user_id: str, user_type: Optional[USER_TYPE]
+    ) -> None: ...
+    def _revoke_roles_from_user(
+        self, roles: Union[str, List[str]], user_id: str, user_type: Optional[USER_TYPE]
+    ) -> None: ...
 
-class _UsersOIDC(_OIDCExecutor[ConnectionSync]):
-    @overload
```

Comment:
the overloads seem to be gone, can you add them back? I think they disappeared in the async refactoring 🤔